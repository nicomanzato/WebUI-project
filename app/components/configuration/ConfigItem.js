'use strict';

import React from 'react';
import {StyleSheet, Text, View, CheckBox} from 'react-native';

class ConfigItem extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      value: this.props.value,
    }

    this.handleOnValueChange = this.handleOnValueChange.bind(this);
  }

  handleOnValueChange() {
    this.setState({
      value: !this.state.value,
    });

    this.props.onChange();
  }

  render() {
    return (
     <View style={styles.configItem}>
       <View style={styles.configTextView}>
         <Text style={styles.configText}>{this.props.title}</Text>
       </View>
       <View style={styles.checkboxView}>
         <CheckBox
          style={styles.configCheckBox}
          onValueChange={this.handleOnValueChange}
          value={this.state.value}
         />
       </View>
     </View>
    );
  }
}

export default ConfigItem;

const styles = StyleSheet.create({
  configItem: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderColor: "#ccc",
    padding: 10,
  },
  configText: {
    fontSize: 17,
  },
  configCheckBox: {},
  configTextView: {
    flex: 0.9,
    height: 50,
  },
  checkboxView: {
    flex: 0.1,
    height: 50,
  }
});
