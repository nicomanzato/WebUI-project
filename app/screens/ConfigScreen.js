import React from 'react';
import { StyleSheet, Text, View, Button, CheckBox } from 'react-native';
import PropTypes from 'prop-types';
import {StackNavigator} from 'react-navigation';

import ConfigList from './../components/configuration/ConfigList';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {
  configToggleVerifiedOnly,
  configToggleDoNotFollow,
  configToggleHaveDefaultInformation,
  configToggleContainsLink,
  configToggleTextTruncated,
} from '../store/configuration/configurationActions';

export const ConfigScreen = (props) => (
  <View style={styles.container}>
    <ConfigList
      configVerifiedOnly={props.configVerifiedOnly}
      configDoNotFollow={props.configDoNotFollow}
      configHaveDefaultInformation={props.configHaveDefaultInformation}
      configContainsLink={props.configContainsLink}
      configTextTruncated={props.configTextTruncated}

      configToggleVerifiedOnly={props.configToggleVerifiedOnly}
      configToggleDoNotFollow={props.configToggleDoNotFollow}
      configToggleHaveDefaultInformation={props.configToggleHaveDefaultInformation}
      configToggleContainsLink={props.configToggleContainsLink}
      configToggleTextTruncated={props.configToggleTextTruncated}
      />
  </View>
);

ConfigScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
};

ConfigScreen.navigationOptions = {

};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFF'
  },
});

function mapStateToProps(state, props) {
  return {
    configVerifiedOnly: state.ConfigurationReducer.configVerifiedOnly,
    configDoNotFollow: state.ConfigurationReducer.configDoNotFollow,
    configHaveDefaultInformation: state.ConfigurationReducer.configHaveDefaultInformation,
    configContainsLink: state.ConfigurationReducer.configContainsLink,
    configTextTruncated: state.ConfigurationReducer.configTextTruncated,
  }
}

const mapDispatchToProps = {
  configToggleVerifiedOnly: () => configToggleVerifiedOnly(),
  configToggleDoNotFollow: () => configToggleDoNotFollow(),
  configToggleHaveDefaultInformation: () => configToggleHaveDefaultInformation(),
  configToggleContainsLink: () => configToggleContainsLink(),
  configToggleTextTruncated: () => configToggleTextTruncated(),
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfigList);
