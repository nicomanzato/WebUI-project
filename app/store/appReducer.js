'use strict';

import { combineReducers } from 'redux'
import PostReducer from './post/postReducer'
import TrendReducer from './trend/trendReducer'
import UserReducer from './user/userReducer'
import nav from './navigation/navReducer'
import ConfigurationReducer from './configuration/configurationReducer'

const AppReducer = combineReducers({
  PostReducer,
  TrendReducer,
  UserReducer,
  nav,
  ConfigurationReducer,
});

export default AppReducer;
