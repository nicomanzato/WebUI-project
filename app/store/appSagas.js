import { call, all, put, takeEvery, takeLatest, take, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'

import postSagas from './post/postSagas'
import trendSagas from './trend/trendSagas'
import userSagas from './user/userSagas'

export default function* appSagas() {
  yield all([
    postSagas(),
    trendSagas(),
    userSagas(),
  ])
}
