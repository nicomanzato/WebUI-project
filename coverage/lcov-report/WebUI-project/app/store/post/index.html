<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/post</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/post
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">92.02% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>173/188</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.56% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>40/41</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">97.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>42/43</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.8% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>168/183</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="postActions.js"><a href="postActions.js.html">postActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="57" class="abs high">57/57</td>
	</tr>

<tr>
	<td class="file high" data-value="postReducer.js"><a href="postReducer.js.html">postReducer.js</a></td>
	<td data-value="95.24" class="pic high"><div class="chart"><div class="cover-fill" style="width: 95%;"></div><div class="cover-empty" style="width:5%;"></div></div></td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="21" class="abs high">20/21</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="95.24" class="pct high">95.24%</td>
	<td data-value="42" class="abs high">40/42</td>
	</tr>

<tr>
	<td class="file medium" data-value="postSagas.js"><a href="postSagas.js.html">postSagas.js</a></td>
	<td data-value="79.03" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 79%;"></div><div class="cover-empty" style="width:21%;"></div></div></td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="93.75" class="pct high">93.75%</td>
	<td data-value="16" class="abs high">15/16</td>
	<td data-value="79.03" class="pct medium">79.03%</td>
	<td data-value="62" class="abs medium">49/62</td>
	</tr>

<tr>
	<td class="file high" data-value="postSelector.js"><a href="postSelector.js.html">postSelector.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="27" class="abs high">27/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Dec 27 2018 17:02:42 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
