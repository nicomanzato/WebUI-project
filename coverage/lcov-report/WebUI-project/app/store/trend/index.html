<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for WebUI-project/app/store/trend</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> WebUI-project/app/store/trend
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/30</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">86.67% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>26/30</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="trendActions.js"><a href="trendActions.js.html">trendActions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="trendReducer.js"><a href="trendReducer.js.html">trendReducer.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="trendSagas.js"><a href="trendSagas.js.html">trendSagas.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Dec 26 2018 13:28:33 GMT-0300 (GMT-03:00)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
