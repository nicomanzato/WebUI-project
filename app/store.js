import { applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga'
import appReducer from './store/appReducer'; //Import the reducer
import appSagas from './store/appSagas';
import reactotron from '../ReactotronConfig'

const sagaMonitor = reactotron.createSagaMonitor()


const sagaMiddleware = createSagaMiddleware({ sagaMonitor })

export const store = reactotron.createStore(appReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(appSagas)
