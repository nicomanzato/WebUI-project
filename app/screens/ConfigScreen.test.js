import React from 'react';
import {ConfigScreen} from './ConfigScreen'
import ReduxWrappedConfigScreen from './ConfigScreen'
import { Provider } from 'react-redux';
import {store} from './../store'
import { shallow, mount, ReactWrapper } from 'enzyme';
import ConfigItem from './../components/configuration/configItem'

describe('config screen', () => {

  it('should render correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <ConfigScreen
        navigation={mockNavigationProp}
        configVerifiedOnly={false}
        configDoNotFollow={false}
        configHaveDefaultInformation={false}
        configContainsLink={false}
        configTextTruncated={false}

        configToggleVerifiedOnly={() => {}}
        configToggleDoNotFollow={() => {}}
        configToggleHaveDefaultInformation={() => {}}
        configToggleContainsLink={() => {}}
        configToggleTextTruncated={() => {}}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = shallow(
      <ReduxWrappedConfigScreen
        navigation={mockNavigationProp}
        store={store}
      />
     );
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle dispatching its actions', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const configToggleVerifiedOnly = jest.fn().mockName('mockedFunction');
    const configToggleDoNotFollow = jest.fn().mockName('mockedFunction');
    const configToggleHaveDefaultInformation = jest.fn().mockName('mockedFunction');
    const configToggleContainsLink = jest.fn().mockName('mockedFunction');
    const configToggleTextTruncated = jest.fn().mockName('mockedFunction');
    const wrapper = mount(
      <ConfigScreen
        navigation={mockNavigationProp}
        configVerifiedOnly={false}
        configDoNotFollow={false}
        configHaveDefaultInformation={false}
        configContainsLink={false}
        configTextTruncated={false}

        configToggleVerifiedOnly={configToggleVerifiedOnly}
        configToggleDoNotFollow={configToggleDoNotFollow}
        configToggleHaveDefaultInformation={configToggleHaveDefaultInformation}
        configToggleContainsLink={configToggleContainsLink}
        configToggleTextTruncated={configToggleTextTruncated}
      />
     );

     const configItems = wrapper.find('ConfigItem');
     configItems.forEach((element) => {
       element.props().onChange();
     });

     expect(configToggleVerifiedOnly).toHaveBeenCalled();
     expect(configToggleDoNotFollow).toHaveBeenCalled();
     expect(configToggleHaveDefaultInformation).toHaveBeenCalled();
     expect(configToggleContainsLink).toHaveBeenCalled();
     expect(configToggleTextTruncated).toHaveBeenCalled();
  });

  it('should call redux wrapped dispatch actions', () => {
    const mockNavigationProp = {state: {params: {data: '1'}}}
    const wrapper = mount(
      <ReduxWrappedConfigScreen
        navigation={mockNavigationProp}
        store={store}
      />
     );

     const configItems = wrapper.find('ConfigItem');
     configItems.forEach((element) => {
       element.props().onChange();
     });

     wrapper.update();

     expect(wrapper).toMatchSnapshot();

  });

});
