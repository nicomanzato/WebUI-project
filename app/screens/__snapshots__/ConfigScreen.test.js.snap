// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config screen should call redux wrapped dispatch actions 1`] = `
<Connect(ConfigList)
  navigation={
    Object {
      "state": Object {
        "params": Object {
          "data": "1",
        },
      },
    }
  }
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConfigList
    configContainsLink={true}
    configDoNotFollow={true}
    configHaveDefaultInformation={true}
    configTextTruncated={true}
    configToggleContainsLink={[Function]}
    configToggleDoNotFollow={[Function]}
    configToggleHaveDefaultInformation={[Function]}
    configToggleTextTruncated={[Function]}
    configToggleVerifiedOnly={[Function]}
    configVerifiedOnly={true}
    navigation={
      Object {
        "state": Object {
          "params": Object {
            "data": "1",
          },
        },
      }
    }
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <Component>
      <View>
        <Component
          style={
            Object {
              "backgroundColor": "#627B97",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 3,
              "marginTop": 1,
              "paddingLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#627B97",
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 3,
                "marginTop": 1,
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Silence notifications from ... 
              </Text>
            </Text>
          </View>
        </Component>
        <ConfigItem
          onChange={[Function]}
          title="People who have not verified the account"
          value={true}
        >
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#ccc",
                  "flexDirection": "row",
                  "padding": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flex": 0.9,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.9,
                      "height": 50,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 17,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      People who have not verified the account
                    </Text>
                  </Text>
                </View>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 0.1,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.1,
                      "height": 50,
                    }
                  }
                >
                  <UnimplementedView
                    onValueChange={[Function]}
                    style={Object {}}
                    value={false}
                  >
                    <Component
                      style={
                        Array [
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "red",
                            "borderWidth": 1,
                          },
                          Object {},
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "flex-start",
                              "borderColor": "red",
                              "borderWidth": 1,
                            },
                            Object {},
                          ]
                        }
                      />
                    </Component>
                  </UnimplementedView>
                </View>
              </Component>
            </View>
          </Component>
        </ConfigItem>
        <ConfigItem
          onChange={[Function]}
          title="People who do not follow"
          value={true}
        >
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#ccc",
                  "flexDirection": "row",
                  "padding": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flex": 0.9,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.9,
                      "height": 50,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 17,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      People who do not follow
                    </Text>
                  </Text>
                </View>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 0.1,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.1,
                      "height": 50,
                    }
                  }
                >
                  <UnimplementedView
                    onValueChange={[Function]}
                    style={Object {}}
                    value={false}
                  >
                    <Component
                      style={
                        Array [
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "red",
                            "borderWidth": 1,
                          },
                          Object {},
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "flex-start",
                              "borderColor": "red",
                              "borderWidth": 1,
                            },
                            Object {},
                          ]
                        }
                      />
                    </Component>
                  </UnimplementedView>
                </View>
              </Component>
            </View>
          </Component>
        </ConfigItem>
        <ConfigItem
          onChange={[Function]}
          title="People who have default profile information"
          value={true}
        >
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#ccc",
                  "flexDirection": "row",
                  "padding": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flex": 0.9,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.9,
                      "height": 50,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 17,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      People who have default profile information
                    </Text>
                  </Text>
                </View>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 0.1,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.1,
                      "height": 50,
                    }
                  }
                >
                  <UnimplementedView
                    onValueChange={[Function]}
                    style={Object {}}
                    value={false}
                  >
                    <Component
                      style={
                        Array [
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "red",
                            "borderWidth": 1,
                          },
                          Object {},
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "flex-start",
                              "borderColor": "red",
                              "borderWidth": 1,
                            },
                            Object {},
                          ]
                        }
                      />
                    </Component>
                  </UnimplementedView>
                </View>
              </Component>
            </View>
          </Component>
        </ConfigItem>
        <ConfigItem
          onChange={[Function]}
          title="Tweets that contain a link"
          value={true}
        >
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#ccc",
                  "flexDirection": "row",
                  "padding": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flex": 0.9,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.9,
                      "height": 50,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 17,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      Tweets that contain a link
                    </Text>
                  </Text>
                </View>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 0.1,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.1,
                      "height": 50,
                    }
                  }
                >
                  <UnimplementedView
                    onValueChange={[Function]}
                    style={Object {}}
                    value={false}
                  >
                    <Component
                      style={
                        Array [
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "red",
                            "borderWidth": 1,
                          },
                          Object {},
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "flex-start",
                              "borderColor": "red",
                              "borderWidth": 1,
                            },
                            Object {},
                          ]
                        }
                      />
                    </Component>
                  </UnimplementedView>
                </View>
              </Component>
            </View>
          </Component>
        </ConfigItem>
        <ConfigItem
          onChange={[Function]}
          title="Tweets that have text truncated"
          value={true}
        >
          <Component
            style={
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#ccc",
                "flexDirection": "row",
                "padding": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "borderBottomWidth": 1,
                  "borderColor": "#ccc",
                  "flexDirection": "row",
                  "padding": 10,
                }
              }
            >
              <Component
                style={
                  Object {
                    "flex": 0.9,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.9,
                      "height": 50,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "fontSize": 17,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      Tweets that have text truncated
                    </Text>
                  </Text>
                </View>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 0.1,
                    "height": 50,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 0.1,
                      "height": 50,
                    }
                  }
                >
                  <UnimplementedView
                    onValueChange={[Function]}
                    style={Object {}}
                    value={false}
                  >
                    <Component
                      style={
                        Array [
                          Object {
                            "alignSelf": "flex-start",
                            "borderColor": "red",
                            "borderWidth": 1,
                          },
                          Object {},
                        ]
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignSelf": "flex-start",
                              "borderColor": "red",
                              "borderWidth": 1,
                            },
                            Object {},
                          ]
                        }
                      />
                    </Component>
                  </UnimplementedView>
                </View>
              </Component>
            </View>
          </Component>
        </ConfigItem>
      </View>
    </Component>
  </ConfigList>
</Connect(ConfigList)>
`;

exports[`config screen should render correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFF",
      "flex": 1,
    }
  }
>
  <ConfigList
    configContainsLink={false}
    configDoNotFollow={false}
    configHaveDefaultInformation={false}
    configTextTruncated={false}
    configToggleContainsLink={[Function]}
    configToggleDoNotFollow={[Function]}
    configToggleHaveDefaultInformation={[Function]}
    configToggleTextTruncated={[Function]}
    configToggleVerifiedOnly={[Function]}
    configVerifiedOnly={false}
  />
</Component>
`;

exports[`config screen should render correctly 2`] = `
<ConfigList
  configContainsLink={false}
  configDoNotFollow={false}
  configHaveDefaultInformation={false}
  configTextTruncated={false}
  configToggleContainsLink={[Function]}
  configToggleDoNotFollow={[Function]}
  configToggleHaveDefaultInformation={[Function]}
  configToggleTextTruncated={[Function]}
  configToggleVerifiedOnly={[Function]}
  configVerifiedOnly={false}
  navigation={
    Object {
      "state": Object {
        "params": Object {
          "data": "1",
        },
      },
    }
  }
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
